### ISSUE TITLE
Failed to build MyPackage with Swift 5.0 for iOS 

### ISSUE THREAD

**Issue by denniscm190** [SwiftPackageIndex/SwiftPackageIndex-Server#1098]
Hello, 

I'm wondering if someone could give an idea of how to solve the following issue.

Recently I published my [Swift package](https://github.com/denniscm190/StockCharts) to Swift Package Index and I noticed that the Build System shows incompatibility with iOS and compatibility with macOS (which is not correct) as you can see in the [Swift Package Index website ](https://swiftpackageindex.com/denniscm190/StockCharts). 

- [x] My swift package works perfectly in iOS. Why is showing incompatibility?

This is the error that I get
```
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64 ${cwd}/Sources/StockCharts/LineChart/LineChartView.swift
	CompileSwift normal arm64 ${cwd}/Sources/StockCharts/LineChart/Helpers/ChartLabel.swift
	CompileSwift normal arm64 ${cwd}/Sources/StockCharts/LineChart/Helpers/LinePath.swift
	CompileSwift normal arm64 ${cwd}/Sources/StockCharts/LineChart/Helpers/LineView.swift
	CompileSwift normal arm64 ${cwd}/Sources/StockCharts/LineChart/Helpers/IndicatorPoint.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(6 failures)
FAILURE 5.0 ios
```

The complete error message is much longer, If you need it just tell me and I will post it.

Any help will be appreciated.

**Comment by finestructure** [comment]
Hi Dennis, thanks for raising an issue.

There appear to be a few different problems at play.

I can see in the following in the error logs for the 5.0 build on iOS:

```
${cwd}/Sources/StockCharts/LineChart/LineChartView.swift:8:8: error: no such module 'SwiftUI'
import SwiftUI
```

Now SwiftUI requires Swift 5.1 or higher so that failure isn’t surprising. What’s surprising is that it passes for macOS with Swift 5.0.

We might be discovering and building a different scheme here.

Another issue I see when looking at the Swift 5.1/iOS build is:

```
error: No signing certificate "iOS Development" found: No "iOS Development" signing certificate matching team ID "MTX83R5H8X" with a private key was found. (in target 'StockCharts' from project 'StockCharts')
```

We won’t be able to build the project if a team id is set. You can see the exact build command we’re using at the top of the build details page. For instance:

```
env DEVELOPER_DIR="/Applications/Xcode_11.3.1.app" xcrun xcodebuild -IDEClonedSourcePackagesDirPathOverride="$PWD/.dependencies" -derivedDataPath "$PWD/.derivedData" build -scheme "StockCharts" -destination "generic/platform=ios"
```

It would be interesting to know if these look correct and work for you for the platforms/Swift versions you expect. Note that you might want to try this with a new user account so the build command doesn’t have access to any credentials, similar to what our builder sees.

Please let me know if that helps and I’ll also try and run the build commands.

**Comment by denniscm190** [comment]
Thanks for your quick reply, Sven. With your explanation I understand better now. So the main issue is that the server can't build the project because It doesn't have access to my signing certificate. So, how should I proceed? Need I to change some settings in the project? Sorry for my low knowledge in this topic, I'm a newbie and I'm still learning. 

Thank you.

**Comment by denniscm190** [comment]
I was able to change the settings to skip code signing. Now, compatibility information is correct. For someone with the same issue check this [answer](https://stackoverflow.com/questions/31039513/how-can-i-skip-code-signing-for-development-builds-in-xcode)

**Comment by finestructure** [comment]
Hi Dennis, great to hear you've managed to solve the issue and thanks for confirming it's ok!

**Comment by finestructure** [comment]
Actually, I do believe there _is_ a remaining issue and I'll re-open this to investigate. My understanding is your package is for iOS and using SwiftUI.

We're showing Swift 5.0 and 4.2 compatibility, which can't be correct. I believe this comes from builds passing for macOS with these Swift versions. That doesn't seem right.

**Comment by finestructure** [comment]
Running this for 5.0/macos-xcodebuild I see this warning before the compile:

```
warning: The iOS deployment target is set to 14.1, but the range of supported deployment target versions for this platform is 8.0 to 12.4.99. (in target 'StockCharts')

Build system information
warning: None of the architectures in ARCHS (x86_64) are valid for this target. Consider setting ARCHS to $(ARCHS_STANDARD) or updating it to include at least one value from VALID_ARCHS (arm64, arm64e, armv7, armv7s). (in target 'StockCharts')
```

yet it happily goes on to compile _something_ - essentially a couple of c files. That's why it's not failing on `import SwiftUI`.

This is what this should look like (taken from the 5.4 build):

```
Build settings from command line:
    ARCHS = x86_64

2021-05-17 08:27:00.607 xcodebuild[23969:8792882]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-18108/DTDeviceKitBase/DTDKRemoteDeviceData.m:371
Details:  (null) deviceType from 00008027-001869C80E84002E was NULL when -platform called.
Object:   <DTDKMobileDeviceToken: 0x1481a1ab0>
Method:   -platform
Thread:   <NSThread: 0x14e015a80>{number = 4, name = (null)}
Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
xcodebuild: error: Failed to build project StockCharts with scheme StockCharts.
	Reason: The run destination My Mac is not valid for Running the scheme 'StockCharts'.
```

@daveverwer , can you tell what's going on with the 5.0 build? I can't run it on my fancy M1 😄  and I can't tell from the command line build. There must be something weird going on with the targets when running on 5.0/4.2.

**Comment by daveverwer** [comment]
> @daveverwer , can you tell what's going on with the 5.0 build? I can't run it on my fancy M1 😄 and I can't tell from the command line build. There must be something weird going on with the targets when running on 5.0/4.2.

I'm afraid not. At least not without installing Catalina as 5.0 and 4.2 both need that.

My guess is the easiest way to do this is to run it manually on the Catalina builder.

**Comment by finestructure** [comment]
Fixed by https://gitlab.com/finestructure/swiftpackageindex-builder/-/merge_requests/65

**Comment by denniscm190** [comment]
Thanks for your help guys. I changed the iOS deployment target version to 12.3 to fix this warning
```
warning: The iOS deployment target is set to 14.1, but the range of supported deployment target versions for this platform is 8.0 to 12.4.99. (in target 'StockCharts')
```
Is everything correct now?

**Comment by finestructure** [comment]
Hi Dennis, thanks for following up! That warning wasn't actually what was causing the remaining issue and it was entirely on our end. I just finished the fix and will deploy it shortly and then re-run the builds for your package. Should be all as expected then, fingers crossed :)

Thanks again for raising the issue, as it exposed a bug in our builder 👌

**Comment by finestructure** [comment]
We're now correctly failing the 5.0 and earlier builds:

<img width="654" alt="CleanShot 2021-05-18 at 11 37 11@2x" src="https://user-images.githubusercontent.com/65520/118629040-7db31880-b7cd-11eb-8736-772c9b307a69.png">

@denniscm190 please note that the iOS builds on main are broken again but this appear to be a genuine failure which I can also reproduce in Xcode 12.5:

```
elpers/ChartLabel.swift:29:38: error: 'Color' is only available in iOS 13.0 or newer
                    .foregroundColor(Color(.systemBlue))
                                     ^
```

I've opened a PR against your repo that should fix it: https://github.com/denniscm190/StockCharts/pull/7
